### DISCUSSION TITLE
Set of `prost` features needed for generated code?

### DISCUSSION THREAD

**Discussion by stefanvanburen** [tokio-rs/prost#1082] [Q&A] [answered]
Hi — I'm generating crates based on code generated with `prost-build`. I'd like to be able to determine the set of features I need to enable in `prost` in order to always successfully compile the generated code — is this available anywhere? Is there a minimum set of features to use, or should I just enable `default-features: true`?

**Comment by caspermeijn** [comment]
The features are documented [here](https://docs.rs/prost/0.13.1/prost/#feature-flags) and [here](https://docs.rs/prost-build/latest/prost_build/#feature-flags).

I am not sure what you are asking. Could you rephrased your question?

**Reply by stefanvanburen** [reply]
Sure, I'll try — imagine I've built some code using `prost-build`, like [on the README](https://github.com/tokio-rs/prost?tab=readme-ov-file#generated-code-example). It'll look something like:

```
#[derive(Clone, PartialEq, ::prost::Message)]
pub struct Person {
    #[prost(string, tag="1")]
    pub name: ::prost::alloc::string::String,
    /// Unique ID number for this person.
    #[prost(int32, tag="2")]
    pub id: i32,
    #[prost(string, tag="3")]
    pub email: ::prost::alloc::string::String,
    #[prost(message, repeated, tag="4")]
    pub phones: ::prost::alloc::vec::Vec<person::PhoneNumber>,
}
```

Clearly, for this code to work, I need `prost` as a dependency, but it's not clear the direct mapping from `prost-build` flags -> `prost` features to enable that will ensure that the generated code compiles.

**Reply by caspermeijn** [reply, marked as answer]
I suggest to use the default feature set, as documented [in the README](https://github.com/tokio-rs/prost?tab=readme-ov-file#using-prost-in-a-cargo-project).

**Reply by stefanvanburen** [reply]
Sure, that's reasonable, and ultimately [what we've done](https://github.com/bufbuild/plugins/blob/62c837b54028866ddeb8f4716cb1fc9024e876b2/plugins/community/neoeinstein-prost/v0.3.1/buf.plugin.yaml#L18-L19). Thanks!

**Reply by caspermeijn** [reply]
Would you mind explaining where you expected this documented? I feel like the readme is complete, but apparently you could not find your answer. Are you open to contributing a PR to improve the documentation?

**Reply by stefanvanburen** [reply]
I guess there's nowhere in the README that directly says, "in order to use the generated code from `prost-build`, you should ensure you've installed `prost` with the default features enabled", or something to that effect.
